rust/agama-utils/src/dbus.rs

```diff
@@ -101,6 +102,21 @@ pub fn extract_id_from_path(path: &OwnedObjectPath) -> Result<u32, zvariant::Err
         })
 }
 
+/// Returns a stream of properties changes.
+///
+/// It listens for changes in several objects that are related to a network device.
+pub async fn build_properties_changed_stream(
+    connection: &zbus::Connection,
+) -> Result<MessageStream, zbus::Error> {
+    let rule = MatchRule::builder()
+        .msg_type(MessageType::Signal)
+        .interface("org.freedesktop.DBus.Properties")?
+        .member("PropertiesChanged")?
+        .build();
+    let stream = MessageStream::for_match_rule(rule, connection, Some(1)).await?;
```

Comment:
the third parameter really needs comment.  I have to google it to found it is some buffer size.